docs/rfc/00-rpc-process.md

```diff
@@ -0,0 +1,52 @@
+author: Nathan Smith <fulcio@nfsmith.ca>
+date: 2022-02-02
+---
+
+# RFC-00 An RFC process
+
+At the moment on the Fulcio project we have a lightweight planning process that
+roughly involves creating an issue on Github for a bug or enhancement and then
+submitting a patch to fix that issue. This works reasonably well for small
+changes, but sometimes a larger change requires more upfront discussion and a
+clear specification for what the finished change will look like. Additionally
+it would be ideal if these important design decision where documented for
+future contributors.
+
+## The RFC process
+
+To create an RFC, create a copy the RFC template to a new file
```

Comment:
I'd suggest we start in a Google doc or hackmd actually, then move to a pull request toward the end.

GitHub PRs are unfortunately really bad for quick, small changes that would probably occur at the start of a design discussion.

The template can still get filled in and sent as a PR, but as a rule of thumb there probably shouldn't be many comments/revisions at that point otherwise the process becomes painful.

Reply:
So to make a concrete suggestion: To create an RFC, open an issue with a link to a google doc or hackmd. After general consensus has been reached, convert to markdown and open a PR in a new file.

You can of course just skip the google doc portion if you're feeling lucky :)

Reply:
Hmmm I just worry that the google docs / hackmd never really make it into the documented code base if that is where they start out. Buttt to be fair, I haven't really participated in this kind of open design work before so I don't know the short coming of Github versus other places to review. Happy to go with prior art or experience. 

The really big thing for me is that we have a pattern for discussing and agreeing on the big work and then making those decisions documented and discoverable after

Reply:
Ah yeah, it's definitely a risk. It's better than not having them though. Starting in code review is pretty hard because small stuff like wordsmithing at the start tends to result in a million comments, and GitHub makes that kind of thing hard to track. With a doc, the reviewer can just fix typos or address small formatting things directly without needing to leave a comment for the author to do it.

Basically I see it as a pre-step, to build consensus and circulate ideas even earlier than getting to a PR/code review stage. Docs also have the benefit that they can have multiple authors, rather than it just being one person submitting a PR being responsible for making all the changes/edits.

But if you want to skip that step for a specific PR/RFC, then I guess you can always go for it!

Reply:
Yeah that totally makes sense! Better to have that first low-barrier idea out there and shared early in easy to collaborate format and open the PR once you're pretty set its a good idea. Ok, I'll write that into the process!